easyaccess/easyaccess.py

```diff
@@ -801,7 +801,25 @@ def query_and_print(self, query, print_time=True,
                 if self.autocommit:
                     self.con.commit()
             print()
-        except:
+        except cx_Oracle.DatabaseError as exc:
+            # Whenever DatabaseError is raised e.g. by a syntax error in a query
+            # close the connection to the database instead of trying to cancel a transaction.
+            # Otherwise, the connection will hang indefinitely during the next attempt to utilize it.
+            # See also:
+            # * https://community.oracle.com/thread/3612149
+            # * https://community.oracle.com/thread/717738
+            # * https://github.com/mgckind/easyaccess/issues/130
+            print(colored('Your query raised DatabaseError:', "red", self.ct))
+            print(colored(str(exc), "red", self.ct))
+            self.con.close()
+            if self.loading_bar:
```

Comment:
I think I have it somewhere, but is the connection reopen after closing it?

Reply:
As far as I checked the next step after closing/canceling connection is pinging the database to check the connection. If the ping comes back as False, the code will connect to database again. Try to run invalid query and then a valid one. 

https://github.com/madamow/easyaccess/blob/318357fbb2f1a56819d16bdc59d1a6032fb30e0f/easyaccess/easyaccess.py#L438-L451